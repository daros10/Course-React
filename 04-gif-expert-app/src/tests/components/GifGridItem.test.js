import React from 'react';
import GifGridItem from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en <GifGridItem />', () => {
  const img = {
    title: 'Dbz',
    url: 'https://dbz.gif',
  };
  const wrapper = shallow(<GifGridItem img={img} />);

  test('Debe mostrar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener el titulo en la etiqueta <p></p>', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(img.title);
  });

  test('Debe obtener una img igual a la url', () => {
    const image = wrapper.find('img');
    expect(image.prop('src')).toBe(img.url);
    expect(image.prop('alt')).toBe(img.title);
  });

  test('El div debe tener animated fadeIn', () => {
    const div = wrapper.find('div');

    expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
  });
});
