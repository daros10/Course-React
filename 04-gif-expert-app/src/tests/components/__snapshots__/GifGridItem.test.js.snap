// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dbz"
    src="https://dbz.gif"
  />
  <p>
    Dbz
  </p>
</div>
`;
